src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureFunction/FunctionDetails.cs

```diff
@@ -0,0 +1,196 @@
+                    {
+                        continue;
+                    }
+
+                    if (!properties.TryGetValue("bindingAttribute", out var triggerAttribute))
+                    {
+                        foreach (var propVal in properties.Values)
+                        {
+                            if (propVal.GetType().Name.Contains("Trigger"))
+                            {
+                                triggerAttribute = propVal;
+                                break;
+                            }
+                        }
+
+                        if (triggerAttribute == null)
+                        {
+                            continue;
+                        }
+                    }
+
+                    var triggerTypeName = triggerAttribute.GetType().Name;
+                    Trigger = triggerTypeName.ResolveTriggerType();
+                    foundTrigger = true;
+                    break;
+                }
+
+                // shouldn't happen, as all functions are required to have a trigger
+                if (!foundTrigger)
+                {
+                    agent.Logger.Debug($"Function {FunctionName} does not have a trigger, defaulting to 'other'");
+                    Trigger = "other";
+                }
+
+                _functionTriggerCache[FunctionName] = Trigger;
+            }
+            else
+            {
+                Trigger = trigger;
+            }
+
+            if (IsWebTrigger)
+            {
+                ParseHttpTriggerParameters(agent, functionContext);
+            }
+        }
+        catch (Exception ex)
+        {
+            agent.Logger.Error(ex, "Error getting Azure Function details.");
+            throw;
+        }
+    }
+
+    private void ParseHttpTriggerParameters(IAgent agent, dynamic functionContext)
+    {
+        if (!_hasAspNetCoreExtensionsReference.HasValue)
+        {
+            // see if the Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore assembly is in the list of loaded assemblies
+            var loadedAssemblies = AppDomain.CurrentDomain.GetAssemblies();
+            var assembly = loadedAssemblies.FirstOrDefault(a => a.GetName().Name == AspNetCoreExtensionsAssemblyName);
+
+            _hasAspNetCoreExtensionsReference = assembly != null;
+
+            if (_hasAspNetCoreExtensionsReference.Value)
+                agent.Logger.Debug($"{AspNetCoreExtensionsAssemblyName} assembly is loaded; InvokeFunctionAsyncWrapper will defer HttpTrigger parameter parsing to FunctionsHttpProxyingMiddlewareWrapper.");
+        }
+
+        // don't parse request parameters here if the Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore assembly is loaded.
+        // If it is loaded, parsing occurs over in FunctionsHttpProxyingMiddlewareWrapper
+        if (_hasAspNetCoreExtensionsReference.Value)
+        {
+            return;
+        }
+
+        object features = functionContext.Features;
```

Comment:
This block of code also needs to look for DT headers - it gets called when the customer is not using the `Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore` package. I believe we have an integration test that exercises this section of code.

Reply:
I will take a look.